src/cpp/session/modules/SessionProfiler.R

```diff
@@ -131,7 +146,34 @@
 
 .rs.addFunction("profilePrint", function(x)
 {
-   .rs.enqueClientEvent("rprof_created", list(
-      path = .rs.scalar(x$x$message$prof_output)
-   ));
+   x$x$message$split <- "h"
+   result <- .rs.rpc.open_profile(list(
+      profvis = x
+   ))
+
+   .rs.enqueClientEvent("rprof_created", result);
+})
+
+.rs.addJsonRpcHandler("clear_profile", function(filePath)
+{
+   tryCatch({
+      resources <- .rs.profileResources()
+
+      filePrefix <- tools::file_path_sans_ext(basename(filePath))
+      
+      profileHtml <- file.path(resources$tempPath, paste(filePrefix, ".html", sep = ""))
+      if (file.exists(profileHtml)) {
+         file.remove(profileHtml)
+      }
+
+      profileDir <- paste(filePrefix, "_files", sep = "")
+      if (file.exists(profileDir)) {
```

Comment:
should this be `file.path(resources$tempPath, profileDir)` so `profileDir` is not resolved against the CWD?